### DISCUSSION TITLE
Nodes Constraints for truss

### DISCUSSION THREAD

**Discussion by mhmdtarhini** [BriefFiniteElementNet/BriefFiniteElement.Net#114] [Q&A] [answered]
Hello,
Why should the support nodes in truss be rotational fixed, aren't they supposed to be only movement fixed ( pinned ) ?
Thank you!

**Comment by epsi1on** [comment, marked as answer]
Hi,
This is because a term named instability. If you do not fix the rotational DoF of a node that is connected to a truss, then the node is unstable, because it can rotate freely. This will make the stiffness matrix to something other than a Positive Definite Matrix and finally probably you'll get a not pos def exception (specially if you use direct solver).

In short because truss element do not resist in rotational direction then nodes that are only connected to truss element, can freely rotate and make model unstable.

There are some code to prevent not pos def exception, which investigates and show some information about what is making the model unstable, read more here:
https://github.com/BriefFiniteElementNet/BriefFiniteElement.Net/wiki/How-to-fix-NotPosDef-error

**Reply by mhmdtarhini** [reply]
Thank you for the fast reply and the information. This library was very useful. Great work! Keep it up!
